platform/default/include/mbgl/storage/offline_schema.sql

```diff
@@ -1,55 +1,153 @@
+
+  accessed INTEGER NOT NULL,                       -- Last time the resource was used by GL Native. Useful for when
+                                                   -- evicting the least used resources from the cache.
+
+  must_revalidate INTEGER NOT NULL DEFAULT 0,      -- When set to true, the resource will not be used unless it gets
+                                                   -- first revalidated by the server.
   UNIQUE (url)
 );
 
+--
+-- Table containing all tiles, both vector and raster.
+--
 CREATE TABLE tiles (
-  id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
-  url_template TEXT NOT NULL,
-  pixel_ratio INTEGER NOT NULL,
-  z INTEGER NOT NULL,
-  x INTEGER NOT NULL,
-  y INTEGER NOT NULL,
-  expires INTEGER,
-  modified INTEGER,
-  etag TEXT,
-  data BLOB,
-  compressed INTEGER NOT NULL DEFAULT 0,
-  accessed INTEGER NOT NULL,
-  must_revalidate INTEGER NOT NULL DEFAULT 0,
+  id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,   -- Primary key.
+
+  url_template TEXT NOT NULL,                      -- The URL of the resource without the access token and without
+                                                   -- the tiles id substituted. Mapbox tiles will be stored using
+                                                   -- the mapbox:// schema. Example:
+                                                   -- mapbox://tiles/user.dataset/{z}/{x}/{y}.vector.pbf
+
+  pixel_ratio INTEGER NOT NULL,                    -- The tile pixel ratio, tipically 1 for vector tiles.
+
+  z INTEGER NOT NULL,                              -- The zoom level of the tile.
+
+  x INTEGER NOT NULL,                              -- The x position of the tile on the tile grid.
+
+  y INTEGER NOT NULL,                              -- The y position of the tile on the tile grid.
+
+  expires INTEGER,                                 -- Expiration time. The tile will be refreshed after this
+                                                   -- expiration is reached. Expired tiles can still be rendered,
+                                                   -- unless must_revalidate is set to true. If an expired tile
+                                                   -- gets rendered, it will be replaced by a newer version as soon
+                                                   -- as the network request with a new tile arrives.
+
+  modified INTEGER,                                -- Last time the tile was modified.
+
+  etag TEXT,                                       -- Checksum used for cache optimization. If, when refreshing the
+                                                   -- tile, it matches the etag on the server, the tile will not
+                                                   -- get re-downloaded. See:
+                                                   -- https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag
+
+  data BLOB,                                       -- Contets of the tile PBF.
```

Comment:
typo: Contents

Reply:
What does the tile PBF contain / why is it important?

Reply:
I don't think there is a restriction of what can be stored as a data. Customers can use proprietary format and decode tiles using custom geometry source if needed. Right?